projects/angular/src/data/datagrid/datagrid-column.ts

```diff
@@ -230,17 +227,18 @@ export class ClrDatagridColumn<T = any>
     }
 
     switch (value) {
-      // the Unsorted case happens when the current state is either Asc or Desc
-      default:
-      case ClrDatagridSortOrder.UNSORTED:
-        this._sort.clear();
-        break;
       case ClrDatagridSortOrder.ASC:
         this.sort(false);
         break;
       case ClrDatagridSortOrder.DESC:
         this.sort(true);
         break;
+      // the Unsorted case happens when the current state is neither Asc nor Desc
+      case ClrDatagridSortOrder.UNSORTED:
+      default:
+        this._sort.clear();
+        this._sortDirection = null;
+        break;
```

Comment:
This case should be moved back to it's original position to minimize this diff and make the change more obvious without have to mentally compare the code.

Further, there are actually two places with the unsorted/default case listed first. I have fixed both in a separate PR: #1721.